modPrecompiled/src/org/aion/precompiled/contracts/TXHashContract.java

```diff
@@ -0,0 +1,30 @@
+package org.aion.precompiled.contracts;
+
+import javax.annotation.Nonnull;
+import org.aion.base.type.IExecutionResult;
+import org.aion.vm.AbstractExecutionResult.ResultCode;
+import org.aion.vm.ExecutionContext;
+import org.aion.vm.ExecutionResult;
+import org.aion.vm.IPrecompiledContract;
+
+public class TXHashContract implements IPrecompiledContract {
+    public static final long COST = 20L;
+
+    private final ExecutionContext context;
+
+    public TXHashContract(@Nonnull final ExecutionContext _context) {
+        context = _context;
+    }
+
+    @Override
+    public IExecutionResult execute(byte[] input, long nrgLimit) {
+
+        long nrgLeft = nrgLimit - COST;
```

Comment:
Do you know if we check that nrgLimit >= 0 before running this method?

I don't know if this would ever happen in real life, but if nrgLimit is a very small negative number, then `nrgLimit - COST` can underflow into a positive number and then it will incorrectly pass the `nrgLeft < 0`  check.